diff --color -Nur Mail-SpamAssassin-4.0.0.orig/sa-update.raw Mail-SpamAssassin-4.0.0/sa-update.raw
--- Mail-SpamAssassin-4.0.0.orig/sa-update.raw	2022-12-13 22:03:27.000000000 -0800
+++ Mail-SpamAssassin-4.0.0/sa-update.raw	2022-12-19 12:28:57.058533774 -0800
@@ -361,7 +361,7 @@
 if ($GPG_ENABLED || $opt{'import'}) {
   # find GPG in the PATH
   # bug 4958: for *NIX it's "gpg", in Windows it's "gpg.exe"
-  $GPGPath = 'gpg' . $Config{_exe};
+  $GPGPath = 'gpg2' . $Config{_exe};
   dbg("gpg: Searching for '$GPGPath'");
 
   if ($GPGPath = Mail::SpamAssassin::Util::find_executable_in_env_path($GPGPath)) {
